task/orchestrator.go

```diff
@@ -120,7 +120,7 @@ func (o *Orchestrator) waitForReadiness(ctx context.Context) (err error) {
 	ctx, cancel := context.WithTimeout(ctx, waitForReadinessTimeout) // so we don't wait indefinitely if there's a problem
 	defer cancel()
 
-	watcher, err := fsnotify.NewWatcher()
+	watcher, err := fsnotify.NewBufferedWatcher(100) // use a buffered chan to prevent a possible race condition
```

Comment:
It's possible, I think, for there to be a race condition where the file is created after the `os.Stat` check (if it already exists) and when we begin receiving on the `watcher.Events` channel. Using a buffered channel should prevent that.